core/src/cell.rs

```diff
@@ -208,22 +209,37 @@ impl<'a> CellProvider for OverlayCellProvider<'a> {
 }
 
 pub struct BlockCellProvider<'a> {
-    output_indices: FnvHashMap<H256, usize>,
+    output_indices: FnvHashMap<&'a H256, usize>,
     block: &'a Block,
 }
 
 impl<'a> BlockCellProvider<'a> {
-    pub fn new(block: &'a Block) -> Self {
-        let output_indices = block
+    pub fn new(block: &'a Block) -> Result<Self, UnresolvableError> {
+        let output_indices: FnvHashMap<&'a H256, usize> = block
             .transactions()
             .iter()
             .enumerate()
-            .map(|(idx, tx)| (tx.hash().to_owned(), idx))
+            .map(|(idx, tx)| (tx.hash(), idx))
             .collect();
-        Self {
+
+        for (input_idx, tx) in block.transactions().iter().enumerate() {
+            for input_pt in tx.input_pts_iter() {
+                if let Some(output_idx) = input_pt
+                    .cell
+                    .as_ref()
+                    .and_then(|cell| output_indices.get(&cell.tx_hash))
+                {
+                    if output_idx.cmp(&input_idx) != Ordering::Less {
```

Comment:
why not just `output_idx >= &input_idx`?

Reply:
try to escape from clippy  :see_no_evil: 
https://rust-lang.github.io/rust-clippy/current/index.html#op_ref
actually, `output_idx >= input_idx` don't compile.

Reply:
another way to pass clippy:
```
output_index.gt(&input_idx)
```

Reply:
I think Clippy recommends `*output_idx >= input_idx`